iree/compiler/Conversion/LinalgToLLVM/Attributes.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IREE_COMPILER_CONVERSION_LINALGTOLLVM_ATTRIBUTES_H_
+#define IREE_COMPILER_CONVERSION_LINALGTOLLVM_ATTRIBUTES_H_
+
+#include "llvm/ADT/StringRef.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+/// Attribute on a entry point function that specifies which function computes
+/// the number of workgroups.
+inline llvm::StringRef getNumWorkgroupsFnAttrName() {
```

Comment:
Instead of having an "llvm" and a "spirv", maybe use the same attribute. You can put this in HAL dialect for now?

Reply:
Its useful to have then separated  for now since some attributes (e.g subgroups) wont exist in LLVM/CPU side

Reply:
Actually these are the markers, I think it make-sense to move this to hal the reset of the code is shared between llvm and spirv. I will do this in the next PR that uses codegen & common in spirv side